group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilderTest.java

```diff
@@ -164,37 +164,40 @@ public void prepareMemberAssignment(
         public TargetAssignmentBuilder.TargetAssignmentResult build() {
             TopicsImage topicsImage = topicsImageBuilder.build().topics();
             // Prepare expected member specs.
-            Map<String, AssignmentMemberSpec> memberSpecs = new HashMap<>();
+            Map<String, MemberSubscriptionSpecImpl> memberSubscriptions = new HashMap<>();
+            Map<String, Map<Uuid, Set<Integer>>> targetAssignment = new HashMap<>(members.size());
 
             // All the existing members are prepared.
-            members.forEach((memberId, member) ->
-                memberSpecs.put(memberId, createAssignmentMemberSpec(
+            members.forEach((memberId, member) -> {
+                Assignment assignment = this.targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
+
+                memberSubscriptions.put(memberId, createMemberSubscriptionSpecImpl(
                     member,
-                    targetAssignment.getOrDefault(memberId, Assignment.EMPTY),
-                    topicsImage
-                )
-            ));
+                    topicsImage,
+                    assignment
+                ));
+            });
 
             // All the updated are added and all the deleted
             // members are removed.
             updatedMembers.forEach((memberId, updatedMemberOrNull) -> {
                 if (updatedMemberOrNull == null) {
-                    memberSpecs.remove(memberId);
+                    memberSubscriptions.remove(memberId);
                 } else {
-                    Assignment assignment = targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
+                    Assignment assignment = this.targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
```

Comment:
I suppose that we could remove `this`.

Reply:
I named it the assignment as target assignment first so had to use this, and then didn't revert, thanks for the catch